[MIG][13.0] stock_picking_completion_info

guewen opened the pull request:
Port #693
Depends on https://github.com/OCA/stock-logistics-warehouse/pull/861

grindtildeath committed:
12.0 create module stock picking completion info

grindtildeath committed:
Distinct full order picking

Commit:
[IMP] stock_picking_completion_info: black, isort

Commit:
[MIG] stock_picking_completion_info: Migration to 13.0

grindtildeath commented:
We should probably rework this module to depend on https://github.com/OCA/stock-logistics-warehouse/pull/861

simahawk commented:
@guewen any feedback?

guewen commented:
@simahawk well, @grindtildeath knows better, he wrote this module :)

Commit:
run pre-commit with new prettiers

grindtildeath commented on stock_picking_completion_info/models/stock_picking.py:
```diff
@@ -0,0 +1,86 @@
+
+    @api.depends(
+        "picking_type_id.display_completion_info",
+        "move_lines.move_dest_ids.move_orig_ids.state",
```
Wrong dependency (cf line 53)

grindtildeath committed:
Rely on stock_move_common_dest as dependency

grindtildeath commented:
@guewen Can you please merge https://github.com/guewen/stock-logistics-warehouse/pull/8 ? 

Maybe also add module dependency https://github.com/OCA/stock-logistics-warehouse/pull/861 in the PR description?

Tonow-c2c commented:
@guewen did you enough right to restart travis ?

simahawk commented:
@grindtildeath if you have time and you want to finish this, please take it over because Guewen is on holidays for 1 week :wink:

grindtildeath commented:
@simahawk @guewen Seems we only need a rebase here, and it should be fine...

guewen commented:
Rebased

simahawk commented:
@guewen is :red_circle:

rousseldenis commented on stock_picking_completion_info/models/stock_picking.py:
```diff
@@ -0,0 +1,84 @@
+        if "state" in vals:
+            # invalidate cache, the api.depends do not allow to find all
+            # the conditions to invalidate the field
+            self.env["stock.picking"].invalidate_cache(fnames=["completion_info"])
```
Is it real or is it the behaviour you have in tests ? If it is just in tests, do the invalidate_cache() there and not here.

grindtildeath replied:
I don't have an issue running the tests without this override, @guewen might know better.

guewen replied:
I think it was not only in tests, IIRC the dependencies in `@api.depends()` of the computed field could not be triggered in every case in chained moves. But maybe now that we have `"move_lines.common_dest_move_ids.state",` in the dependencies it's no longer the case and this could be removed?

guewen commented:
@grindtildeath tests fail after de4dcfd, can you have a look?

grindtildeath committed:
Fix tests after change in dependencies

grindtildeath commented:
@guewen should be fixed with https://github.com/guewen/stock-logistics-warehouse/pull/10

Commit:
Fix warning in stock_move_common_dest

In this module: https://github.com/OCA/stock-logistics-warehouse/pull/808
A computed field uses "common_dest_move_ids" as dependency:

    @api.depends(
        "picking_type_id.display_completion_info",
        "move_lines.common_dest_move_ids.state",
    )
    def _compute_completion_info(self):

Which has the following effect when the ORM triggers changes:

odoo.osv.expression: Non-stored field stock.move.common_dest_move_ids cannot be searched.

Implement a search method to prevent this.

jgrandguillaume reviewed (approved):
LGTM

jgrandguillaume commented:
/ocabot merge

guewen commented:
/ocabot merge minor
